pynecone/utils.py

```diff
@@ -582,6 +582,25 @@ def get_api_port() -> int:
     return port
 
 
+def is_process_on_port(port) -> bool:
+    """Check if a process is running on the given port.
+
+    Args:
+        port: The port.
+
+    Returns:
+        Whether a process is running on the given port.
+    """
+    for proc in psutil.process_iter(["pid", "name", "cmdline"]):
```

Comment:
I think we can make a function `get_process_on_port` which does all this logic, and can be reused for the functions `is_process_on_port` and `kill_process_on_port`